sycl/include/sycl/handler.hpp

```diff
@@ -68,7 +68,7 @@
 // 3300+ lines of code
 
 // SYCL_LANGUAGE_VERSION is 4 digit year followed by 2 digit revision
-#if !SYCL_LANGUAGE_VERSION || SYCL_LANGUAGE_VERSION < 202001
```

Comment:
`SYCL_LANGUAGE_VERSION` is a compiler-provided macro, meaning that the only situation this `#if` can be taken is when SYCL headers are used with 3rd-party host compiler. I don't know why would we need a different API in that case - most likely this condition is some kind of legacy code and I would completely drop `__SYCL_NONCONST_FUNCTOR__` and all its uses (meaning `_KERNELFUNCPARAM_TYPE` as well).